tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionTranspilerSuite.scala

```diff
@@ -383,6 +397,18 @@ class RegularExpressionTranspilerSuite extends FunSuite with Arm {
     assertCpuGpuMatchesRegexpFind(patterns, inputs)
   }
 
+  test("fall back to CPU for newline next to line or string anchor") {
+    // these patterns were discovered during fuzz testing and resulted in different
+    // results between CPU and GPU
+    val patterns = Seq(raw"\w[\r,B]\Z", raw"\s\Z\Z", "^$\\s", "$x*\\r", "$\\r")
```

Comment:
`\\r` is an escaped 'r' which isn't a valid regexp. `\r` is the carriage return 
```suggestion
    val patterns = Seq(raw"\w[\r,B]\Z", raw"\s\Z\Z", "^$\\s", "$x*\r", "$\r")
```

Reply:
These patterns were generated by the fuzz tests and resulted in different behavior between CPU and GPU so I think we need to make sure we fall back to CPU for these patterns.

Reply:
On second thought I think you are right. I think I ended up with an extra `\` in here from copy/paste.